examples/src/index.js

```diff
@@ -81,14 +82,16 @@ function Demo() {
 
   return (
     <div>
+      
       <Segment>
     <Grid columns={2}>
       <Grid.Column>
         <MarkdownAsInputEditor readOnly={true} plugins={plugins} markdown={markdown} onChange={onMarkdownChange}/>
       </Grid.Column>
 
       <Grid.Column>
-        <SlateAsInputEditor readOnly={false} lockText={true} plugins={plugins} value={slateValue} onChange={onSlateValueChange} editorProps={propsObj} />
+        <Button onClick={() => setDisplay(!display)}>TOGGLE READONLY</Button>
+        <SlateAsInputEditor key={display} readOnly={display} lockText={true} plugins={plugins} value={slateValue} onChange={onSlateValueChange} editorProps={propsObj} />
```

Comment:
I don't understand why we need both `key` and `readOnly` if they have the same value.

Reply:
I am also confused by this. A `key` is usually required when creating a list of React elements, where each element has a unique `key` attribute (https://reactjs.org/docs/lists-and-keys.html). I do not understand the way it is being used here.

Reply:
@dselman & @DianaLease A `Key` need not have to be `{display}` but it can be `Math.random()` also, the problem is if we do not give a key which changes on the button click, the child component's toolbar does not rerender. So, a key should be provided which can be `display` (Changes true and false on a button click), `Math.Random()` (generates a random number every time button is click) or even `new Date()` (generates new date every time a button is clicked). 

To rerender the child component we need to have a key which changes on button click @irmerk.

Reply:
@codingdefined This feels a bit hacky to me. The last sentence of the React docs [here](https://reactjs.org/docs/reconciliation.html) explains, "Keys should be stable, predictable, and unique. Unstable keys (like those produced by Math.random()) will cause many component instances and DOM nodes to be unnecessarily recreated, which can cause performance degradation and lost state in child components."

When the `readOnly` prop changes, it will cause a re-render of the `SlateAsInputEditor` with the new prop. It's possible though that Slate is doing some optimization at the editor level, which we have seen happen in other places. I would try checking the readOnly prop on [this line](https://github.com/accordproject/markdown-editor/blob/f5105bc2d760a0cf70c98f73e8287bb51089586e/src/SlateAsInputEditor/index.js#L415) to conditionally render the Toolbar. The other option I can think of is also passing readOnly as a prop to ToolbarWrapper [here](https://github.com/accordproject/markdown-editor/blob/f5105bc2d760a0cf70c98f73e8287bb51089586e/src/SlateAsInputEditor/index.js#L431) and using the prop in the styled component to toggle a display: none property.

Reply:
@DianaLease None of the two options actually solved the issue. 

And I guess, the keys in this sense is predictable because we are setting it as true or false, though its a hack it might not cause any performance degradation.
cc : @irmerk